### PR TITLE
Java: remove URL

### PR DESCRIPTION
**Pull request by dkropachev**
We will need it later for https://github.com/scylladb/alternator-load-balancing/pull/40

### CODE REVIEW

File: java/src/main/java/com/scylladb/alternator/AlternatorLiveNodes.java [hunk 1]
```diff
@@ -86,10 +82,10 @@ public URI nextAsURI() {
...
 
-    public URL nextAsURL(String file) {
+    public URI nextAsURI(String file) {
         try {
-            return new URL(alternatorScheme, nextNode(), alternatorPort, file);
-        } catch (MalformedURLException e) {
+            return new URI(alternatorScheme, "",  nextNode(), alternatorPort, file, "", "");
+        } catch (URISyntaxException e) {
             // Can only happen if alternatorScheme is an unknown one.
             logger.log(Level.WARNING, "nextAsURL", e);
```
Comment (Bouncheck) [inline java/src/main/java/com/scylladb/alternator/AlternatorLiveNodes.java:90]: This part and comment should be updated too. URI won't throw if provided with unknown to URL format scheme.

File: java/src/main/java/com/scylladb/alternator/AlternatorLiveNodes.java [hunk 2]
```diff
@@ -105,11 +101,11 @@ private static String streamToString(java.io.InputStream is) {
 
     private void updateLiveNodes() {
         List<String> newHosts = new ArrayList<String>();
-        URL url = nextAsURL("/localnodes");
+        URI uri = nextAsURI("/localnodes");
         try {
             // Note that despite this being called HttpURLConnection, it actually
             // supports HTTPS as well.
-            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+            HttpURLConnection conn = (HttpURLConnection) uri.toURL().openConnection();
```
Comment (Bouncheck) [inline java/src/main/java/com/scylladb/alternator/AlternatorLiveNodes.java:108]: note: Now if nextAsUri is provided with invalid scheme for URL we'll get MalformedURLException here instead. It's handled by catch but just wanted to make sure that's intended.

**Review by Bouncheck — COMMENTED** [review]
LGTM. I defer the approval decision to the maintainer.

**Comment by nyh** [comment]
> LGTM. I defer the approval decision to the maintainer.

Looks to me like you have more experience in this than I do. I don't remember what are the finer differences between URL and URI, nor did I understand what is the motivation of this change (though I can **guess** it has something to do with encoding the parameters, I didn't understand why URL isn't good enough for that).

@Bouncheck I see you left some review comments, please leave an approval when you are ok with the change and I can merge it.

**Comment by dkropachev** [comment]
> > LGTM. I defer the approval decision to the maintainer.
> 
> Looks to me like you have more experience in this than I do. I don't remember what are the finer differences between URL and URI, nor did I understand what is the motivation of this change (though I can **guess** it has something to do with encoding the parameters, I didn't understand why URL isn't good enough for that).
> 
> @Bouncheck I see you left some review comments, please leave an approval when you are ok with the change and I can merge it.

URL is not good enough, because we want to encode rack and datacenter parameters properly, and only URI does it.

File: java/src/main/java/com/scylladb/alternator/AlternatorLiveNodes.java (hunk 2 above)
Comment (dkropachev) [reply java/src/main/java/com/scylladb/alternator/AlternatorLiveNodes.java:108]: Thanks for pointing this out, i will address it in next PRs

File: java/src/main/java/com/scylladb/alternator/AlternatorLiveNodes.java (hunk 1 above)
Comment (dkropachev) [reply java/src/main/java/com/scylladb/alternator/AlternatorLiveNodes.java:90]: Thanks, removed the comment.

**Comment by dkropachev** [comment]
@Bouncheck, could you please take a look again.

**Comment by Bouncheck** [comment]
I don't see the new changes pushed

**Comment by dkropachev** [comment]
> I don't see the new changes pushed

done

File: java/src/main/java/com/scylladb/alternator/AlternatorLiveNodes.java (hunk 1 above)
Comment (Bouncheck) [reply java/src/main/java/com/scylladb/alternator/AlternatorLiveNodes.java:90]: nit: `nextAsURL` inside `logger.log(` should change to `nextAsURI`

**Review by Bouncheck — APPROVED** [review]
LGTM, there's one typo.

**Comment by dkropachev** [comment]
> LGTM, there's one typo.

thanks, fixed.

**Pull request merged**